### DISCUSSION TITLE
How to flash a GSI ROM on Lenovo Tab K10!

### DISCUSSION THREAD

**Discussion by CmlCool** [phhusson/treble_experimentations#2654] [Ideas]
Hello everyone, I recently got a Tab K10, so I decided to make a complete guide on installing a gsi ROM.
![lenovo-tab-k10](https://github.com/user-attachments/assets/49006e64-9f5f-4eb7-acd7-b430b693802a)

1. Unlocking the Bootloader. To do this, open the settings app > system > about and tap Build Number 7 times.

2. Go back to system > developer options > and enable OEM Unlocking and USB Debugging.

3. Now connect your tablet to you PC, grab the latest adb and fastboot from [here](https://developer.android.com/tools/releases/platform-tools) , unzip, and in the directory name of where you extracted, type "cmd". In CMD, type `adb reboot bootloader && fastboot flashing unlock`. You will see a prompt on your tablet's screen, press volume up and in cmd type `fastboot reboot` after unlock pass. Go through the initial setup now.

4. Let's begin! power off your tablet, press `Volume Down + Power Button` simultaneously and quickly connect the USB cable. You will see "FASTBOOT MODE..". Now in CMD, type `fastboot reboot fastboot`  and wait until the tablet boots into fastbootd.

5. Now let's flash that GSI! first, type `fastboot delete-logical-partition product_a && fastboot create-logical-partition product_a 1000` to free up System space.

6. Erase the system partition to prepare flashing: type `fastboot erase system`.

7. Flashing the system image. First get one, from [here](https://github-wiki-see.page/m/phhusson/treble_experimentations/wiki/Generic-System-Image-%28GSI%29-list). Type `fastboot flash system PATH_TO_SYSTEM_IMG` and replace "PATH_TO_SYSTEM_IMG" with the directory in which you have the GSI\system.img. 

8. After the flashing is done, it is very important to regenerate the partitions to avoid any bootloops. To do so, type `fastboot -w`.

9. Disabling Dm-Verity, Verifications and flashing Vbmeta. To do so, type `fastboot reboot bootloader && fastboot flash --disable-verity --disable-verification vbmeta PATH_TO_STOCK_VBMETA` in which you replace "PATH_TO_STOCK_VBMETA with the path to the  vbmeta from stock firmware\vbmeta.img.

10. Flashing Stock Boot image. Although it's not really necessary, it's recommended. Type `fastboot reboot fastboot && fastboot flash boot PATH_TO_STOCK_BOOT` in which you replace "PATH_TO_STOCK_BOOT" with path to boot image from stock firmware\boot.img.

11. Almost done! Now using the volume keys, navigate to "Enter Recovery" and Format Data (Wipe data/Factory Reset).

12. Congratulations! Now select reboot system, wait around 5-10 minutes for the tablet to boot and enjoy your brand new, fresh system!

Short update: you may encounter some bugs with your GSI, such as Wifi/LTE not working.

Screenshots of RisingOS running on Tab K10:

![Screenshot_20240927-113126_Ortus Launcher](https://github.com/user-attachments/assets/27280501-5a94-411a-9f1b-d6ab84660d22)
![Screenshot_20240927-113226_Settings](https://github.com/user-attachments/assets/2abd11d2-0239-4df5-9e03-a1b9f66dc701)

**Comment by ida-lovelace** [comment]
There's already a guide for it in the wiki. https://github.com/phhusson/treble_experimentations/wiki/Lenovo-Tab-K10

**Comment by RadioKiller** [comment]
Hi I have a problem when running android 12+ gsi they soft reboot and the browser for some reason refreshes every second and games freeze while playing and I have to reboot the device can u help me fix it?
